src/sysfiles.c warning at compilation

I am not sure if this is an error or not. When I compile most recent master with 
````
./autogen.sh
./configure
make
````
then I get the following warning message:
````
   C       src/sysfiles.c => obj/sysfiles.lo
src/sysfiles.c: In function ‘SyExecuteProcess’:
src/sysfiles.c:2990:30: warning: variable ‘func2’ might be clobbered by ‘longjmp’ or ‘vfork’ [-Wclobbered]
     sig_handler_t           *func2;
                              ^
````
This does not seem to harm the compilation process, and all other messages indicate that the compilation ended successfully. And indeed, gap can be started, no problems there. 

So I am unsure if this needs to be addressed or not. My system is debian jessie x64. I can provide the complete outputs if necessary.

--- a/src/sysfiles.c
+++ b/src/sysfiles.c
@@ -2987,7 +2987,7 @@ UInt SyExecuteProcess (
     Int                     tin;                    /* temp in             */
     Int                     tout;                   /* temp out            */
     sig_handler_t           *func;
-    sig_handler_t           *func2;
+    sig_handler_t           * volatile func2;
 
 
     /* turn off the SIGCHLD handling, so that we can be sure to collect this child

